### PR TITLE
Fix unit tests

### PR DESCRIPTION
**Pull request by cmpich**

### CODE REVIEW

File: source/org/zfin/antibody/repository/HibernateAntibodyRepository.java [hunk 1]
```diff
@@ -906,81 +919,59 @@ private void populateAntibodyStatisticsRecordPaginated(GenericTerm aoTerm, Antib
...
-        criteria.add(Restrictions.eq("geno.wildtype", true));
-        criteria.add(Restrictions.eq("fishox.standardOrGenericControl", true));
-        criteria.add(Restrictions.or(Restrictions.eq("xpatres.entity.superterm", term),
-            Restrictions.eq("xpatres.entity.subterm", term)));
-
         if (withImgOnly) {
-            criteria.add(Restrictions.isNotEmpty("images"));
+            hqlClauses.add("size(figure.images) > 0");
+        }
+        if (CollectionUtils.isNotEmpty(hqlClauses)) {
```
Comment (rtaylorzfin) [inline source/org/zfin/antibody/repository/HibernateAntibodyRepository.java:962]: Another place that could be:
```
hql += hqlClauses.isEmpty() ? "" : " where " + String.join(" and ", hqlClauses);
```
or if you don't need the empty check:
```
hql += " where " + String.join(" and ", hqlClauses);
```

File: source/org/zfin/expression/repository/HibernateExpressionRepository.java [hunk 2]
```diff
@@ -2056,147 +2055,104 @@ public List<GenericTerm> getWildTypeAnatomyExpressionForMarker(String zdbID) {
...
-            aliasMap.put("exp.experimentConditions", "cond");
-            aliasMap.put("cond.conditionDataType", "cdt");
-            criteria.add(Restrictions.eq("cdt.name", "heat shock"));
-            logger.debug("chemical environments only");
-        }
-
-        //now add all of the aliases that were marked as necessary above
-        for (Map.Entry<String, String> entry : aliasMap.entrySet()) {
-            criteria.createAlias(entry.getKey(), entry.getValue());
+        if (CollectionUtils.isNotEmpty(hqlClauses)) {
```
Comment (rtaylorzfin) [inline source/org/zfin/expression/repository/HibernateExpressionRepository.java:2136]: one more:
```
hql += hqlClauses.isEmpty() ? "" : " where " + String.join(" and ", hqlClauses);
```

**Review by rtaylorzfin — APPROVED** [review]
Looks good

File: source/org/zfin/antibody/repository/HibernateAntibodyRepository.java [hunk 3]
```diff
@@ -230,26 +229,40 @@ OR exists ( select 1 from TransitiveClosure child
...
+
+        hqlClauses.add("expressionExperiment.antibody = :antibody");
+        parameterMap.put("antibody", antibody);
+        hqlClauses.add("(expressionResult.superTerm = :term OR expressionResult.subTerm = :term )");
+        parameterMap.put("term", aoTerm);
+        hqlClauses.add("size(publication.figures) > 0");
+        hqlClauses.add("expressionResult.expressionFound = true");
+        hqlClauses.add("fishExperiment.standardOrGenericControl = true");
+        hqlClauses.add("fishExperiment.fish.genotype.wildtype = true");
+        if (CollectionUtils.isNotEmpty(hqlClauses)) {
```
[the remark this answers is not in the document]
Comment (cmpich) [reply source/org/zfin/antibody/repository/HibernateAntibodyRepository.java:253]: Good catch! WIll change. I had copied this from a previous change.

**Pull request merged**